net/rpc/server.go

```diff
@@ -483,7 +516,7 @@ func (server *Server) ServeRequest(codec ServerCodec) error {
 	}
 
 	handler := func() error {
-		return service.call(server, sending, nil, mtype, req, argv, replyv, codec)
+		return service.call(ctx, server, sending, nil, mtype, req, argv, replyv, codec)
```

Comment:
While we are doing this shouldn't pass the context to the various interceptors? That could be useful to pass data to interceptors.

Reply:
That could be not possible though because the context is optional, but maybe we can default to a background context in that case 🤔

Reply:
Passing it to the interceptors would also break the signature, so we'd have to go from 2 interceptors to 4 (with 2 being deprecated).

Perhaps we can revisit when more of this is practically usable?